storage/common/include/algebra/storage/matrix.hpp

```diff
@@ -0,0 +1,454 @@
+/** Algebra plugins, part of the ACTS project
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+#pragma once
+
+// @TODO: Remove this when Vc fixes their false positives.
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic warning "-Wdeprecated-declarations"
+#endif
```

Comment:
I'm very baffled by this. This also seems to enable a warning, not disable it. Am I reading this wrong?